From a325e4253c854826adc5735a40e12440dc016043 Mon Sep 17 00:00:00 2001
From: Zhantao Tang <zhantao.tang@windriver.com>
Date: Mon, 18 Oct 2021 19:05:04 +0800
Subject: [PATCH] pkcs11-hse: Makefile: using internal compile variables in
 bitbake system

In bsp31 rc3, the pkcs11-hse demo changed its make file and set the
CC variable to "$(CROSS_COMPILE)gcc", this will make the demo build
failure due to CROSS_COMPILE is not set in bitbake system, so this
patch will replace it with internal CC variable in bitbake system.
So as LD variable. And change the message "CROSS_COMPILE is not set"
to be a warning to make the compile to continue.

Upstream-Status: Pending

Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>

%% original patch: 0001-pkcs11-hse-Makefile-using-internal-compile-variables.patch

squash! pkcs11-hse: Makefile: using internal compile variables in bitbake system
---
 Makefile                                         | 6 +++---
 examples/hse-encrypt/Makefile                    | 4 ++--
 examples/hse-secboot/Makefile                    | 2 +-
 examples/hse-sysimg/Makefile                     | 4 ++--
 examples/pkcs-key-provision/Makefile             | 4 ++--
 examples/pkcs-keyop/Makefile                     | 4 ++--
 examples/trustzone-key-provision/target/Makefile | 4 ++--
 7 files changed, 14 insertions(+), 14 deletions(-)

diff --git a/Makefile b/Makefile
index 0618b58..1627093 100644
--- a/Makefile
+++ b/Makefile
@@ -19,7 +19,7 @@ endif
 
 # Prerequisites check
 ifeq (,$(CROSS_COMPILE))
-    $(error CROSS_COMPILE is not set)
+    $(warning CROSS_COMPILE is not set)
 endif
 
 ifeq (,$(HSE_FWDIR))
@@ -37,8 +37,8 @@ INSTALL_LIBDIR := $(INSTALL_DIR)/lib
 INSTALL_BINDIR := $(INSTALL_DIR)/bin
 
 # Build libraries
-CC := $(CROSS_COMPILE)gcc
-LD := $(CROSS_COMPILE)ld
+CC := $(CC)
+LD := $(LD)
 CFLAGS ?= -fPIC -Wall -g
 LDFLAGS ?=
 
diff --git a/examples/hse-encrypt/Makefile b/examples/hse-encrypt/Makefile
index 695cade..c82978f 100644
--- a/examples/hse-encrypt/Makefile
+++ b/examples/hse-encrypt/Makefile
@@ -7,10 +7,10 @@ include ../common.mk
 all: hse-encrypt
 
 hse-encrypt: hse-encrypt.c
-	$(CROSS_COMPILE)gcc $(LDFLAGS) $(INCLUDE_LIBHSE) $(CFLAGS) $^ -o $@ $(LD_LIBHSE)
+	$(CC) $(LDFLAGS) $(INCLUDE_LIBHSE) $(CFLAGS) $^ -o $@ $(LD_LIBHSE)
 
 clean:
 	rm -f hse-encrypt
 
 install:
-	install hse-encrypt $(EXAMPLES_INSTALLDIR)
\ No newline at end of file
+	install hse-encrypt $(EXAMPLES_INSTALLDIR)
diff --git a/examples/hse-secboot/Makefile b/examples/hse-secboot/Makefile
index e3e1b31..a518710 100644
--- a/examples/hse-secboot/Makefile
+++ b/examples/hse-secboot/Makefile
@@ -7,7 +7,7 @@ include ../common.mk
 all: hse-secboot
 
 hse-secboot: hse-secboot.c
-	$(CROSS_COMPILE)gcc $(LDFLAGS) -I$(OPENSSL_DIR)/include $(INCLUDE_LIBHSE) $(CFLAGS) $^ -o $@ \
+	$(CC) $(LDFLAGS) -I$(OPENSSL_DIR)/include $(INCLUDE_LIBHSE) $(CFLAGS) $^ -o $@ \
 	$(LD_OPENSSL) $(LD_LIBHSE)
 
 clean:
diff --git a/examples/hse-sysimg/Makefile b/examples/hse-sysimg/Makefile
index 3c36395..5fd4454 100644
--- a/examples/hse-sysimg/Makefile
+++ b/examples/hse-sysimg/Makefile
@@ -7,10 +7,10 @@ include ../common.mk
 all: hse-sysimg
 
 hse-sysimg: hse-sysimg.c
-	$(CROSS_COMPILE)gcc $(LDFLAGS) $(INCLUDE_LIBHSE) $(CFLAGS) $^ -o $@ $(LD_LIBHSE)
+	$(CC) $(LDFLAGS) $(INCLUDE_LIBHSE) $(CFLAGS) $^ -o $@ $(LD_LIBHSE)
 
 clean:
 	rm -f hse-sysimg
 
 install:
-	install hse-sysimg $(EXAMPLES_INSTALLDIR)
\ No newline at end of file
+	install hse-sysimg $(EXAMPLES_INSTALLDIR)
diff --git a/examples/pkcs-key-provision/Makefile b/examples/pkcs-key-provision/Makefile
index daed62f..da890a1 100644
--- a/examples/pkcs-key-provision/Makefile
+++ b/examples/pkcs-key-provision/Makefile
@@ -7,10 +7,10 @@ include ../common.mk
 all: pkcs-key-provision
 
 pkcs-key-provision: pkcs-key-provision.c
-	$(CROSS_COMPILE)gcc $(LDFLAGS) $(INCLUDE_LIBPKCS) $(CFLAGS) $^ -o $@ $(LD_OPENSSL) -ldl
+	$(CC) $(LDFLAGS) $(INCLUDE_LIBPKCS) $(CFLAGS) $^ -o $@ $(LD_OPENSSL) -ldl
 
 clean:
 	rm -f pkcs-key-provision
 
 install:
-	install pkcs-key-provision $(EXAMPLES_INSTALLDIR)
\ No newline at end of file
+	install pkcs-key-provision $(EXAMPLES_INSTALLDIR)
diff --git a/examples/pkcs-keyop/Makefile b/examples/pkcs-keyop/Makefile
index 3a68f42..88c6aca 100644
--- a/examples/pkcs-keyop/Makefile
+++ b/examples/pkcs-keyop/Makefile
@@ -10,10 +10,10 @@ INCLUDE_KEYOP ?= -I$(OPENSSL_DIR)/include \
 all: pkcs-keyop
 
 pkcs-keyop: pkcs-keyop.c
-	$(CROSS_COMPILE)gcc $(LDFLAGS) $(INCLUDE_KEYOP) $(CFLAGS)  $^ -o $@ $(LD_OPENSSL) $(LD_LIBP11)
+	$(CC) $(LDFLAGS) $(INCLUDE_KEYOP) $(CFLAGS)  $^ -o $@ $(LD_OPENSSL) $(LD_LIBP11)
 
 clean:
 	rm -f pkcs-keyop
 
 install:
-	install pkcs-keyop $(EXAMPLES_INSTALLDIR)
\ No newline at end of file
+	install pkcs-keyop $(EXAMPLES_INSTALLDIR)
diff --git a/examples/trustzone-key-provision/target/Makefile b/examples/trustzone-key-provision/target/Makefile
index f592afc..eabba04 100644
--- a/examples/trustzone-key-provision/target/Makefile
+++ b/examples/trustzone-key-provision/target/Makefile
@@ -15,10 +15,10 @@ TEEC_INCLUDE := -I$(TEEC_EXPORT)/include
 all: hse-kek-provision hse-symkey-provision
 
 hse-kek-provision: hse-kek-provision.c
-	$(CROSS_COMPILE)gcc $(LDFLAGS) -L$(LIBHSE_DIR) $(INCLUDE_LIBHSE) $(INCDIR) $(CFLAGS) $^ -o $@ -lhse
+	$(CC) $(LDFLAGS) -L$(LIBHSE_DIR) $(INCLUDE_LIBHSE) $(INCDIR) $(CFLAGS) $^ -o $@ -lhse
 
 hse-symkey-provision: hse-symkey-provision.c
-	$(CROSS_COMPILE)gcc $(LDFLAGS) $(INCLUDE_LIBHSE) $(INCDIR) $(TEEC_INCLUDE) \
+	$(CC) $(LDFLAGS) $(INCLUDE_LIBHSE) $(INCDIR) $(TEEC_INCLUDE) \
 	$(CFLAGS) $^ -o $@ $(TEEC_LIB)
 
 clean:
-- 
2.25.1

